### PR TITLE
(CDAP-5074) Make RunId available through RuntimeContext

### PR DESCRIPTION
**Pull request by chtyim**
- Also move the scala compiler plugin management to top level pom

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/RuntimeContext.java [hunk 1]
```diff
@@ -39,4 +40,9 @@
    * @return The application namespace
    */
   String getNamespace();
+
```
Comment (sreevatsanraman) [inline cdap-api/src/main/java/co/cask/cdap/api/RuntimeContext.java]: Are we returning Twill Runids or the ones that is generated in CDAP. Maybe we should just expose the one that is generated in CDAP?

Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/RuntimeContext.java]: It is always the CDAP runID. The Twill runId never surface to the user.

**Comment by sreevatsanraman** [comment]
LGTM

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT3609

**Pull request merged**